accessible/toolbox_treeview.component.js

```diff
@@ -0,0 +1,145 @@
+/**
+ * AccessibleBlockly
+ *
+ * Copyright 2016 Google Inc.
+ * https://developers.google.com/blockly/
+ *
+ * Licensed under the Apache License, Version 2.0 (the 'License');
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an 'AS IS' BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @fileoverview Angular2 Component that details how blocks are
+ * rendered in the toolbox in AccessibleBlockly. Also handles any interactions
+ * with the blocks.
+ * @author madeeha@google.com (Madeeha Ghori)
+ */
+blocklyApp.ToolboxTreeView = ng.core
+  .Component({
+    selector: 'toolbox-tree-view',
+    template: `
+      <li #parentList [ngClass]="{blocklyHasChildren: displayBlockMenu || block.inputList.length > 0, blocklyActiveDescendant: index == 0 && tree.getAttribute('aria-activedescendant') ==
+            'blockly-toolbox-tree-node0'}" [attr.aria-selected]="index == 0 && tree.getAttribute('aria-activedescendant') ==
+            'blockly-toolbox-tree-node0'" role="treeitem" [attr.aria-level]="level" id="{{getCategoryId(index, parentList)}}">
+        {{setActiveDesc(parentList)}}
+        <label #blockSummaryLabel id="blockly-{{block.id}}" style="color:red">{{block.toString()}}</label>
```

Comment:
So, if line 30 looks like so:
`<li #parentList [attr.aria-labelledby]="utilsService.concatStringWithSpaces('blockly-block-summary', blockSummaryLabel.id)"...`
and line 35 is commented out, I get [this error.](http://imgur.com/kvWutYY)
TLDR: the function concatStringWithSpaces returns two different results when called right after each other.

I believe this is because when the HTML is being loaded, it first creates the parentList element and sets the aria-labelledby based on the blockSummary.id (which is null because the blockSummary element hasn't been created yet.). Then, when the labelledby function is called a second time, the blockSummary element *has* been created. That's why I'm setting labelledby after the blockSummary element has been created.